app/services/pda/provider_details.rb

```diff
@@ -109,5 +110,23 @@ def response
     def query_api
       conn.get url
     end
+
+    def contact_id
+      response = conn.get("#{Rails.configuration.x.pda.url}/provider-users/#{encoded_username}")
+      if response.success?
+        if response.status == 200
+          JSON.parse(response.body).dig("user", "ccmsContactId")
+        else
+          Rails.logger.info("#{self.class} - No provider details found for #{@provider.username}")
+          raise ValidDetailsNotFound, "No provider details found for #{@provider.username}"
+        end
+      else
+        raise ApiError, "API Call Failed: provider-users (#{response.status}) #{response.body}"
+      end
+    end
```

Comment:
rather than reusing a response var/method (could confuse things) I would write separate methods for each and use a base_url for the faraday connection